Helper/Entity/ProductHelper.php

```diff
@@ -459,6 +473,42 @@ private function getSubProducts(Product $product)
         return $subProducts;
     }
 
+    /**
+     * Returns all parent product IDs, e.g. when simple product is part of configurable or bundle
+     *
+     * @param array $productIds
+     * @return array
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function getParentProductIds(array $productIds)
+    {
+        $parentIds = [];
+        foreach ($this->getCompositeTypes() as $typeInstance) {
+            $parentIds = array_merge($parentIds, $typeInstance->getParentIdsByChild($productIds));
+        }
+
+        return $parentIds;
+    }
+
+    /**
+     * Returns composite product type instances
+     *
+     * @return AbstractType[]
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @see \Magento\Catalog\Model\Indexer\Product\Flat\AbstractAction::_getProductTypeInstances
+     */
+    private function getCompositeTypes()
+    {
+        if (null === $this->compositeTypes) {
```

Comment:
Could you swap it to `$this->compositeTypes === null`? We keep this convention all around the extension.

Reply:
Absolutely, I prefer it as well